src/ray/core_worker/core_worker.cc

```diff
@@ -4015,13 +4015,16 @@ void CoreWorker::HandleExit(rpc::ExitRequest request,
   // any object pinning RPCs in flight.
   bool is_idle = !own_objects && pins_in_flight == 0;
   bool force_exit = request.force_exit();
-  RAY_LOG(DEBUG) << "Exiting: is_idle: " << is_idle << " force_exit: " << force_exit;
+  auto worker_id = WorkerID::FromBinary(request.worker_id());
+
+  RAY_LOG(DEBUG) << "Exiting: worker_id" << worker_id << " is_idle: " << is_idle << " force_exit: " << force_exit;
   if (!is_idle && force_exit) {
     RAY_LOG(INFO) << "Force exiting worker that owns object. This may cause other "
                      "workers that depends on the object to lose it. "
                   << "Own objects: " << own_objects
                   << " # Pins in flight: " << pins_in_flight;
   }
+  core_worker_client_pool_->Disconnect(worker_id);
```

Comment:
I don't think this fixes the memory leak since the worker process will exit after this.

Reply:
Can you see if my current logic is correct?...this Exit will be called by remote worker exit. And in my debug code `[2023-11-30 00:02:48,167 D 2887 2920] core_worker_client_pool.cc:42: 1 size Connected to xxx:11935` size of the map keep 1.